include/nbl/builtin/hlsl/memory_accessor.hlsl

```diff
@@ -25,33 +25,59 @@ struct MemoryAdaptor
     }
 
     template<typename Scalar>
-    void get(const uint ix, NBL_REF_ARG(Scalar) value) { accessor.get(ix, value);}
+    enable_if_t<sizeof(Scalar) == sizeof(uint32_t), void> get(const uint ix, NBL_REF_ARG(Scalar) value) 
+    { 
+        uint32_t aux;
+        accessor.get(ix, aux);
+        value = bit_cast<Scalar, uint32_t>(aux);   
+    }
     template<typename Scalar>
-    void get(const uint ix, NBL_REF_ARG(vector <Scalar, 2>) value) { accessor.get(ix, value.x), accessor.get(ix + _NBL_HLSL_WORKGROUP_SIZE_, value.y);}
+    enable_if_t<sizeof(Scalar) == sizeof(uint32_t), void> get(const uint ix, NBL_REF_ARG(vector <Scalar, 2>) value) 
+    {
+        uint32_t2 aux;
+        accessor.get(ix, aux.x);
+        accessor.get(ix + _NBL_HLSL_WORKGROUP_SIZE_, aux.y);
+        value = bit_cast<vector<Scalar, 2>, uint32_t2>(aux);
+    }
```

Comment:
you can go harder and more compact
```cpp
template<typename Scalar, uint16_t Dims>
enable_if_t<sizeof(Scalar)==sizeof(uint32_t) && Dims>1 && Dims<=4,void> get(const uint32_t ix, NBL_REF_ARG(vector<Scalar,Dims>) value)
```

and then use a for loop conditioned on `Dims` 

P.S. I don't know if a `is_same_v<vector<T,1>,T>` in HLSL hence the case only for Dim>1

Reply:
Checked, `vector<T,1>` is not just a typedef for T, because it supports `operator[]` and the `.x` swizzle